pymilvus/milvus_client/milvus_client.py

```diff
@@ -0,0 +1,801 @@
+                not included in the result data.
+        """
+
+        # TODO: Figure out thread safety
+        # with self.concurrent_lock:
+        #     self.concurrent_counter += 1
+
+        if self.collection is None:
+            logger.error("Collection does not exist: %s", self.collection_name)
+            raise ValueError(
+                "Missing collection. Make sure data inserted or intialized on existing collection."
+            )
+
+        if not isinstance(data[0], list):
+            data = [data]
+
+        return_fields = list(self.fields.keys())
+        return_fields.remove(self.vector_field)
+
+        res = self.collection.search(
+            data,
+            self.vector_field,
+            expr=filter_expression,
+            param=search_params or self.default_search_params,
+            limit=top_k,
+            partition_names=partitions,
+            output_fields=return_fields,
+            timeout=timeout or self.timeout,
+        )
+
+        ret = []
+        for hits in res:
+            for hit in hits:
+                ret_dict = {x: hit.entity.get(x) for x in return_fields}
+                ret.append({"score": hit.score, "data": ret_dict})
+
+        # TODO: Figure out thread safety
+        # with self.concurrent_lock:
+        #     self.concurrent_counter -= 1
+        return ret
+
+    def query_data(
+        self,
+        filter_expression: str,
+        partitions: List[str] = None,
+        timeout: int = None,
+    ) -> List[dict]:
+        """Query for entries in the Collection.
+
+        Args:
+            filter_expression (str): The filter to use for the query.
+            partitions (List[str], optional): Which partitions to perform query. Defaults to None.
+            timeout (int, optional): Timeout to use, overides the client level assigned at init.
+                Defaults to None.
+
+        Raises:
+            ValueError: Missing collection.
+
+        Returns:
+            List[dict]: A list of result dicts, embeddings are not included.
+        """
+
+        # TODO: Figure out thread safety
+        # with self.concurrent_lock:
+        #     self.concurrent_counter += 1
+
+        if self.collection is None:
+            logger.error("Collection does not exist: %s", self.collection_name)
+            raise ValueError(
+                "Missing collection. Make sure data inserted or intialized on existing collection."
+            )
+
+        return_fields = list(self.fields.keys())
+        return_fields.remove(self.vector_field)
+
+        res = self.collection.query(
+            expr=filter_expression,
+            partition_names=partitions,
+            output_fields=return_fields,
+            timeout=timeout or self.timeout,
+        )
+
+        # TODO: Figure out thread safety
+        # with self.concurrent_lock:
+        #     self.concurrent_counter -= 1
+
+        return res
+
+    def get_embeddings_by_pk(
```

Comment:
Great wrapper for the `get_embeddings_by_pk` and `delete_by_pk`. Without this, I noticed that there are many issues in the community related to how to get vectors by pk and delete by pk. I believe this will make these developers' life easier.